.circleci/config.yml

```diff
@@ -276,16 +285,46 @@ jobs:
           name: Run unit tests
           command: |
             SCCACHE_CACHE_SIZE=2G
-            cargo test --package << parameters.crate >> --all-features --lib -- --nocapture
+            cargo test -p << parameters.crate >> --all-features --lib -- --nocapture
       - run:
           name: Run integration tests
-          # Only run integration tests if there are any
           command: |
             SCCACHE_CACHE_SIZE=2G
             set +o pipefail
-            (cargo test --package << parameters.crate >> --all-features --test '*' -- --list 2>&1 | grep -q "no test target matches pattern") \
-              && echo "nothing to test" \
-              || cargo test --package << parameters.crate >> --all-features --test '*' -- --nocapture
+            path=$(echo "<< parameters.crate >>" | sed "s/shuttle-\(.*\)/\1/")
+            if [ -f "$path/integration.sh" ]; then
+              bash "$path/integration.sh"
+            else
+              echo "nothing to test here, adjust CI file or add an integration script"
+              exit 1
+            fi
+      - save-cargo-cache
+  test-workspace-member-with-integration-docker:
+    parameters:
+      crate:
+        description: "Crate in workspace to test"
+        type: string
+    # Using a machine image since tests will start a docker container
+    executor: machine-ubuntu
+    steps:
+      - install-rust
+      - install-protoc
+      - checkout
+      - run: git submodule update --init
+      - restore-cargo-cache
+      - apply-patches
+      - run:
+          name: Run integration tests that need docker
+          command: |
+            SCCACHE_CACHE_SIZE=2G
+            set +o pipefail
+            path=$(echo "<< parameters.crate >>" | sed "s/shuttle-\(.*\)/\1/")
+            if [ -f "$path/integration_docker.sh" ]; then
+              bash "$path/integration_docker.sh"
+            else
+              echo "nothing to test here, adjust CI file or add an integration script"
+              exit 1
+            fi
```

Comment:
It might be, but I've opted to do it this way so that
(1): You can easily run the same command locally.
(2): If you need to customize the ordering of tests execution, or anything else, per crate.

Reply:
I would take simplicity anytime (a one-liner based on `cargo`) :), but I don't have a strong preference 

(1) I think we run locally just `cargo tests --all-features`. Also, the same command ran by the CI can be found in the config.yaml. 
(2) Sounds like an overoptimisation. Do we need to customize anything when running tests right now?

Reply:
> Also, the same command ran by the CI can be found in the config.yaml.

That is the more annoying option. We want it to be easy for both dev and CI to run.

> Do we need to customize anything when running tests right now?

Only the builder exception right now, but I made this change because I suspect that we will need some sort of customization of test running to reduce spurious fails.